libsrc/cbm/tgi_lo.inc

```diff
@@ -0,0 +1,488 @@
+;
+; Graphics driver for a max. 160x50x2 (but 4 pixels) mode on all CBMs.
+;
+; 2018-08-30, Stefan Haubenthal
+;
+
+        .include        "zeropage.inc"
+
+        .include        "tgi-kernel.inc"
+        .include        "tgi-error.inc"
+        .include        "cbm_kernal.inc"
```

Comment:
I must admit that I don't have the overview here. Is this line to be changed?

And maybe there's a bigger problem behind it:

I seem to understand that you need to call into those Kernal wrapper thingies on the Plus/4. However that code is part of the C library and .o65 based driver can't just call into the C library as the cc65 .o65 loader doesn't do any resolving of "outgoing" symbol references.

The usual two options to solve this is to either duplicate the code in question from the C library into the driver or use the libref mechanism: The cc65 driver loader makes the address of the libref symbol available to the loaded driver. That libref symbol is the same for all drivers of a certain type (TGI, SER, JOY, ...) on a certain target. However, as there's - at least currently - only one TGI driver for the Plus/4 I don't see a problem with this driver making use of the Plus/4 tgi_libref as it sees fit. Technically one would create some jump table like e.g. in https://github.com/cc65/cc65/blob/master/libsrc/atari/serref.s

As there's a dummy tgi_libref present on every target this approach doesn't keep the Plus/4 driver from working on other CBM machines. It just must make sure at runtime to not make use of the jump table.

Reply:
Exactly, I don't want cbm_kernal.inc anymore, so I'll use libref.

Reply:
I see. However, the libref overhead might be larger than a duplication of the functionality in the driver if it's only about hitting some softswitch, doing the Kernal JSR and hitting some other softswitch...